tests/unit/elements/UserElementTest.php

```diff
@@ -81,6 +81,27 @@ public function testValidateUnverifiedEmail(): void
         $this->tester->deleteElement($user);
     }
 
+    public function testUniqueAttributesForInactiveUsers(): void
+    {
+        $user = new User([
+            'active' => true,
+            'email' => $this->inactiveUser->email,
+            'username' => $this->inactiveUser->username,
+            'unverifiedEmail' => $this->inactiveUser->unverifiedEmail,
+        ]);
+
+        $this->tester->saveElement($user);
+
+        self::assertFalse($user->hasErrors());
+
+        Craft::$app->getUsers()->activateUser($this->inactiveUser);
+
+        self::assertTrue($this->inactiveUser->hasErrors('email'));
+        self::assertTrue($this->inactiveUser->hasErrors('username'));
+
```

Comment:
Inactive user can now not be activated because there is an active user with the same email/username.